wayshot/src/cli.rs

```diff
@@ -51,4 +54,17 @@ pub struct Cli {
     /// Present a fuzzy selector for output/display selection
     #[arg(long, alias = "chooseoutput", conflicts_with_all = ["slurp", "output"])]
     pub choose_output: bool,
+
+    /// Path to your config file.
+    /// Defaults to:
+    ///     1. `$XDG_CONFIG_HOME/wayshot/config.toml`
+    ///     2. `$HOME/wayshot/config.toml` -- if `$XDG_CONFIG_HOME` variable doesn't exist
+    ///     3. `None` -- if the config isn't found, the `Config::default()` will be used
```

Comment:
I'm presuming `Config::default` is the current functionality inorder to not make it a breaking change?
Simply put, with the patch merged, wayshot will act just as it did but will now respect an optional configuration?

Reply:
With exception of some breaking changes (big comment below), yes.

As for precedence, it can be described as such: `cli` arguments > `cli.config` config's values > `$XDG_CONFIG_HOME/wayshot/config.toml` values (doesn't exist by default) > `$HOME/wayshot/config.toml` values (doesn't exist by default) > `Config::default()` values (hard-coded to have same behavior as `cli` default arguments). The repo's `config.toml` example file will also have the same default values, as current `cli` ones.